src/components/TokenEnterAmount.test.tsx

```diff
@@ -0,0 +1,219 @@
+import { fireEvent, render } from '@testing-library/react-native'
+import React from 'react'
+import { getNumberFormatSettings } from 'react-native-localize'
+import { Provider } from 'react-redux'
+import { AmountEnteredIn } from 'src/send/types'
+import { StoredTokenBalance, TokenBalance } from 'src/tokens/slice'
+import { NetworkId } from 'src/transactions/types'
+import { createMockStore } from 'test/utils'
+import {
+  mockCeloTokenBalance,
+  mockCeloTokenId,
+  mockEthTokenId,
+  mockTestTokenTokenId,
+  mockTokenBalances,
+} from 'test/values'
+import TokenEnterAmount, { APPROX_SYMBOL } from './TokenEnterAmount'
+
+jest.mock('react-native-localize')
+
+const getMockStoreTokenBalances = (): Record<string, StoredTokenBalance> => ({
+  ...mockTokenBalances,
+  [mockCeloTokenId]: {
+    ...mockTokenBalances[mockCeloTokenId],
+    priceUsd: '0.5',
+    balance: '10',
+  },
+  [mockEthTokenId]: {
+    ...mockTokenBalances[mockEthTokenId],
+    tokenId: mockEthTokenId,
+    balance: '10',
+    networkId: NetworkId['ethereum-sepolia'],
+    showZeroBalance: true,
+    isNative: true,
+    symbol: 'ETH',
+    priceFetchedAt: Date.now(),
+    name: 'Ether',
+  },
+  [mockTestTokenTokenId]: {
+    // token with no price
+    balance: '10',
+    tokenId: mockTestTokenTokenId,
+    networkId: NetworkId['celo-alfajores'],
+    showZeroBalance: false,
+    isNative: false,
+    symbol: 'TST',
+    name: 'Test Token',
+    decimals: 18,
+    address: '0xtest',
+  },
+})
+const mockStoreTokenBalances = getMockStoreTokenBalances()
+const mockStore = { tokens: { tokenBalances: mockStoreTokenBalances } }
+
+describe('TokenEnterAmount', () => {
+  beforeEach(() => {
+    jest.clearAllMocks()
+    jest
+      .mocked(getNumberFormatSettings)
+      .mockReturnValue({ decimalSeparator: '.', groupingSeparator: ',' })
+  })
+
+  const mockOnInputChange = jest.fn()
+  const mockToggleAmountType = jest.fn()
+  const mockOnOpenTokenPicker = jest.fn()
+
+  const defaultProps = {
+    token: {
+      ...mockCeloTokenBalance,
+    } satisfies TokenBalance,
+    inputValue: '1',
+    tokenAmount: '1',
+    localAmount: '$0.1',
+    amountType: 'token' as AmountEnteredIn,
```

Comment:
@kathaypacific considering this is more for the intellisense purpose I would keep it as is!

Reply:
i'm not sure i understand - explicit casting like this is generally dangerous because `'blah' as AmountEnteredIn` will not produce any errors whereas `'blah' as const` will

Reply:
@kathaypacific but calling intellisense on the empty string will provide you available options of `token` and `local`

Reply:
i see, so we are prioritising the experience of writing this value. my concern comes down to `as SomeType` is a way to ignore all the static type checking functionality that Typescript gives us, so I try to avoid it. if the allowed values of `AmountEnteredIn` changes then this test will not be flagged, but i guess at runtime the tests will fail (slower time to uncover the problem). we've been quite inconsistent with casting using the type or `const` though so it's also fine by me to keep it like this